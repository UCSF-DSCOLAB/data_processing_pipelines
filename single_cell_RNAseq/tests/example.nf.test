nextflow_pipeline {

    name "Test Workflow pipeline_pre_qc.nf"
    script "pipeline_pre_qc.nf"
    tag "qc-quantiles"


    def c4PathToTestFiles = "/krummellab/data1/integration_test_user/tutorial_lib_sep"

    // Assertion directories
    def finding_doublets = "/finding_doublets"
    def automated_processing = "/automated_processing"
    def freemuxlet = "/freemuxlet"
    def test_directories = [finding_doublets, automated_processing, freemuxlet]

    stage {
        copy c4PathToTestFiles
    }

    test("Should successfully demultiplex merged libaries using freemuxlet and run doublet finder") {

        def metaDir = "$metaDir"
        def testDirAbsolute = metaDir + c4PathToTestFiles
        def freemuxDir = testDirAbsolute + "/freemuxlet_data"

        when {
            params {
                    load("./tests/inputs/pre_qc.json")
                    project_dir = testDirAbsolute
                    settings = {
                        add_tcr = false
                        add_bcr = false
                        skip_cellranger = true
                        merge_for_demux = true
                        merge_demux_dir = freemuxDir
                        demux_method = "freemuxlet"
                        run_doubletfinder = true
                        mincell = 3
                        minfeature = 100
                        default_qc_cuts_dir = testDirAbsolute
                        default_qc_cuts_file = "default_qc_cuts.csv"
                        randomseed = 21212
                        remove_demux_DBL = true
                        remove_all_DBL = true
                    }
            }
        }

        then {
            // Setup assertion directories
            def dm1_scg1 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM1-SCG1"
            def dm1_scg2 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM1-SCG2"
            def dm2_scg1 = testDirAbsolute + "/data/single_cell_GEX/processed/TEST-POOL-DM2-SCG1"
            def test_sample_paths = [dm1_scg1, dm1_scg2, dm2_scg1]
            def resultFilesList = []
            for (sample in test_sample_paths) {
                for (test_dir in test_directories) {
                    def dir_to_test = sample + test_dir
                    def dirFile = new File(dir_to_test)
                    def resultFiles = dirFile.listFiles().collect { it.name }
                    resultFilesList.addAll(resultFiles)
                    println "Processing: ${resultFilesList}"
                }
            }
            // Filter out the .pdf files
            def filteredPaths = resultFilesList.findAll { filePath ->
                     !filePath.endsWith('.pdf')
            }
            def sorted = filteredPaths.sort()
            assert snapshot(workflow, sorted).match()
        }
    }
}

