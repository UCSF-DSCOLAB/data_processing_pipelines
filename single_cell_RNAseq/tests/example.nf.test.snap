{
    "Should successfully demultiplex merged libaries using freemuxlet and run doublet finder": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    "Pipeline completed at: 2024-07-31T14:28:38.975357401-07:00",
                    "Execution status: OK",
                    "Deleting working directory /krummellab/data1/erflynn/data_processing_pipelines/single_cell_RNAseq/.nf-test/tests/3016d8dcd518820bd4edc5469a74d3ca/work"
                ],
                "errorMessage": "",
                "name": "workflow",
                "success": true
            },
            [
                "TEST-POOL-DM1-SCG1_doublet_counts.csv:md5,6ddbe773b71de72a979975ad7ecbfa15",
                "TEST-POOL-DM1-SCG1_doubletStat.rds:md5,2abec2c196a0f2edc675aa54e4ef3e81",
                "TEST-POOL-DM1-SCG1_seurat_object_findingDoublets.rds:md5,58389f8af2a840dbb35c175c7001b990",
                "TEST-POOL-DM1-SCG1_cutoffs.csv:md5,44e63748a1f261f028c093a463d1a09d",
                "TEST-POOL-DM1-SCG1_raw.rds:md5,ae40d8eb177c0ac53e42a907f29e0395",
                "TEST-POOL-DM1-SCG1_quantiles_pre.tsv:md5,94816302c1bce6019a4cfea34e07178b",
                "barcodes_of_interest.filt.list:md5,6cc4789ad17ab2865d026126d1c3eafa",
                "TEST-POOL-DM1-SCG1.var.gz:md5,51a8822ed9db7df403240897aeae3ca0",
                "TEST-POOL-DM1-SCG1.cel.gz:md5,2845a88ef8245e637703afc348eecad2",
                "TEST-POOL-DM1-SCG1.umi.gz:md5,2e092070c4de1bc61446d671ce40e40b",
                "TEST-POOL-DM1-SCG1.plp.gz:md5,90c13506d711e1af548b95a071c588ab",
                "TEST-POOL-DM1-SCG1.clust1.vcf.gz:md5,766ef1c11cbfa0b4ff76b44e79a91438",
                "TEST-POOL-DM1-SCG1.clust1.samples.gz:md5,4b8dcc7f68dfd1d8baadb8d7919b0a37",
                "TEST-POOL-DM1-SCG1.lmix:md5,b7e4b08d9edc41d1e7985d5a37aa1ddd",
                "TEST-POOL-DM1-SCG1.clust1.samples.reduced.tsv:md5,afd48131fde269d3d7fb027e7ef4a0e8",
                "TEST-POOL-DM1-SCG2_seurat_object_findingDoublets.rds:md5,f33357c9281208c39792ceab4a0afb38",
                "TEST-POOL-DM1-SCG2_doubletStat.rds:md5,b3d31f12ee003848c7ef5cd07d08eb21",
                "TEST-POOL-DM1-SCG2_doublet_counts.csv:md5,b379ccdbfc2abda59e35f19d9aada274",
                "TEST-POOL-DM1-SCG2_cutoffs.csv:md5,44e63748a1f261f028c093a463d1a09d",
                "TEST-POOL-DM1-SCG2_raw.rds:md5,c414b0c7ac36a4292b4e445ca11a26ed",
                "TEST-POOL-DM1-SCG2_quantiles_pre.tsv:md5,03c104a82af781e44ded3e03ccd4524c",
                "barcodes_of_interest.filt.list:md5,5eb29c41775ec1d4d59b526b51f41d91",
                "TEST-POOL-DM1-SCG2.cel.gz:md5,cb9abbaa76f309f04d9000bb2fdf3ce1",
                "TEST-POOL-DM1-SCG2.var.gz:md5,4305b2ca04757168697d9b0f9d970b6e",
                "TEST-POOL-DM1-SCG2.umi.gz:md5,cd2942489731eda4bf58ea6069f97750",
                "TEST-POOL-DM1-SCG2.plp.gz:md5,f67566334e86cb73299af708eb1f3a72",
                "TEST-POOL-DM1-SCG2.clust1.samples.gz:md5,84b32b0d91716d02f272991a7973bf6a",
                "TEST-POOL-DM1-SCG2.clust1.samples.reduced.tsv:md5,f9c5e2c01b2aa3eb6e1f7cc8261f95bf",
                "TEST-POOL-DM1-SCG2.lmix:md5,e41a51bf1ea89f855f32b85785c1f8f6",
                "TEST-POOL-DM1-SCG2.clust1.vcf.gz:md5,766ef1c11cbfa0b4ff76b44e79a91438",
                "TEST-POOL-DM2-SCG1_doublet_counts.csv:md5,9a5fcc173622e3d965e276d4af5ab6a9",
                "TEST-POOL-DM2-SCG1_doubletStat.rds:md5,ae65810041d783a0b8d5bcc3e0538bff",
                "TEST-POOL-DM2-SCG1_seurat_object_findingDoublets.rds:md5,326103e3b50fb26953b3f9726a538e25",
                "TEST-POOL-DM2-SCG1_cutoffs.csv:md5,44e63748a1f261f028c093a463d1a09d",
                "TEST-POOL-DM2-SCG1_raw.rds:md5,4001a01db8a1a700b83bf4920cbb8ced",
                "TEST-POOL-DM2-SCG1_quantiles_pre.tsv:md5,fd9ad3a0d63fdbf72bbe94084ba45531",
                "barcodes_of_interest.filt.list:md5,c35e74b984701342ea0181b8f58135fb",
                "TEST-POOL-DM2-SCG1.umi.gz:md5,e471254f7069f84eaee0bfb67557ebcc",
                "TEST-POOL-DM2-SCG1.plp.gz:md5,4c6ccdc6f7caa45ea03e208d5f18f6d5",
                "TEST-POOL-DM2-SCG1.cel.gz:md5,fecf810b425f99665801e3a3645be777",
                "TEST-POOL-DM2-SCG1.var.gz:md5,61da1ce3dceb89154a567891aafb837b",
                "TEST-POOL-DM2-SCG1.clust1.vcf.gz:md5,d921988704d5dfc21a9dbabc6f182a07",
                "TEST-POOL-DM2-SCG1.clust1.samples.gz:md5,428fa3a68f95f8328a18ca7e02ba9547",
                "TEST-POOL-DM2-SCG1.clust1.samples.reduced.tsv:md5,6c27161f7516e101abe8bf1150c4a564",
                "TEST-POOL-DM2-SCG1.lmix:md5,75d7df0745f9f5d91719fa730f0337de"
            ]
        ],
        "timestamp": "2024-07-31T14:28:39.437523688"
    }
}